Build automation for Fedora

The proposed change
-------------------
Some automation is added for building on Fedora-like systems (ideally Github Actions CI)

Benefits of this change
-----------------------
Contributors on these systems have fewer problems


Any issues this might cause
---------------------------
GH actions only support Ubuntu, so might need to be Docker-in-Docker....

So far, I've got a Dockerfile that builds Fedora 33 with all the dependencies needed to run most of the tests.
I'm hitting an error in the Docker container that I don't get on my laptop, hoping someone can help me troubleshoot.

(I also made sure I have *gvaluetypes.h* in the container, which contains the *G_VALUE_HOLDS_INT* definition, so not sure why I'm getting the `test_undo_space` warning)

```tests/test_qltk_entry.py::TUndoEntry::test_undo_space
  /quodlibet/.venv/lib64/python3.9/site-packages/gi/overrides/Gtk.py:218: Warning: g_value_get_int: assertion 'G_VALUE_HOLDS_INT (value)' failed
    return super(Editable, self).insert_text(text, -1, position)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
========================================================================================================== short test summary info ===========================================================================================================
FAILED tests/test_formats__audio.py::TAudioFile::test_is_writable - AssertionError: True is not false
FAILED tests/test_operon.py::TOperonAdd::test_permissions - AssertionError: False != True : (0, '', '')
FAILED tests/test_operon.py::TOperonPrint::test_permissions - AssertionError: False != True : (0, 'Silence\n', '')
FAILED tests/test_util_atomic.py::Tatomic_save::test_readonly - AssertionError: OSError not raised
FAILED tests/plugin/test_mediaserver.py::TMediaServer::test_entry_name - dbus.exceptions.DBusException: org.freedesktop.DBus.Error.Spawn.ExecFailed: /usr/bin/dbus-launch terminated abnormally without any error message
... (a bunch more related errors)
============================================================================== 33 failed, 4287 passed, 41 skipped, 123 warnings, 53 errors in 114.20s (0:01:54) ==============================================================================

```

I cut out a lot of the extra error messages since I think theyre related to the first ones,
but if it'd be helpful, I can include the entire error output.

Here's the Dockerfile, in case it's helpful
[Dockerfile.fedora33](https://gist.github.com/Phil305/75d1e21299facfe7d7db79e057c58213)

--- a/.github/workflows/fedora33/Dockerfile
+++ b/.github/workflows/fedora33/Dockerfile
@@ -1,0 +1,42 @@
+FROM fedora:33
+
+# DNF dependencies required before poetry can install virtualenv dependencies
+RUN dnf -y install \
+      poetry \
+      cairo-devel \
+      gcc \
+      python3-devel \
+      dbus-devel \
+      gobject-introspection-devel \
+      cairo-gobject-devel
+
+# Dependencies for tests
+RUN dnf -y install \
+      Xvfb \
+      # Contains 'GdkPixbuf' namespace (used by 'gi') \
+      gtk3 \
+      # Contains 'Gst' module \
+      gstreamer1 \
+      # Contains GStreamer replaygain plugin \
+      gstreamer1-plugins-good \
+      # SVG support (for 'Timage_support.test_create_pixbuf' test) \
+      librsvg2 \
+      dbus-x11
+
+RUN dnf clean all
+
+# Create user (required for write access tests in 'tests/test_formats__audio.py' to pass)
+ARG HOST_USER_ID=5555
+ENV HOST_USER_ID ${HOST_USER_ID}
+RUN useradd -u $HOST_USER_ID -ms /bin/bash user
+
+# Required to prevent pytest from running into permissions errors when creating
+# cache files in GITHUB_WORKSPACE as 'user'
+ENV PYTHONDONTWRITEBYTECODE 1
+
+# required for translation tests that use 'gettext'
+ENV LANG C.UTF-8
+
+# run tests
+COPY entrypoint.sh /entrypoint.sh
+ENTRYPOINT ["/entrypoint.sh"]

--- a/.github/workflows/fedora33/action.yml
+++ b/.github/workflows/fedora33/action.yml
@@ -1,0 +1,6 @@
+# action.yml
+name: Fedora 33
+description: Run tests for Fedora 33
+runs:
+  using: docker
+  image: Dockerfile

--- a/.github/workflows/fedora33/entrypoint.sh
+++ b/.github/workflows/fedora33/entrypoint.sh
@@ -1,0 +1,8 @@
+#!/usr/bin/env bash
+
+# Fix for hanging flake8 test
+# https://github.com/quodlibet/quodlibet/issues/3539#issuecomment-767389459
+poetry config virtualenvs.in-project true
+
+poetry install -E plugins
+sudo -u user poetry run python3 setup.py test

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -3,6 +3,14 @@ name: test
 on: [push, pull_request]
 
 jobs:
+  fedora:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Run tests in Docker container
+        uses: ./.github/workflows/fedora33
 
   msys2:
     runs-on: windows-latest

